src/sandstorm/run-bundle.c++

```diff
@@ -1730,11 +1786,21 @@ private:
         KJ_SYSCALL(setenv("MAIL_URL", config.mailUrl.cStr(), true));
       }
       if (config.rootUrl == nullptr) {
-        if (config.port == 80) {
-          KJ_SYSCALL(setenv("ROOT_URL", kj::str("http://", config.bindIp).cStr(), true));
+        kj::String scheme;
```

Comment:
You can use `kj::StringhPtr` here and remove the `kj::str()` wrappers to avoid redundant allocation and copying.